File: core/QingWaTiaoTaiJieWenTiLcof/QingWaTiaoTaiJieWenTiLcof.py
def numWays(n: int) -> int:
    pre, cur = 1, 1
    for i in range(2, n + 1):
        cur, pre = cur + pre, cur

    return cur % 1000000007


if __name__ == '__main__':
    print(numWays(5))

File: core/QingWaTiaoTaiJieWenTiLcof/QingWaTiaoTaiJieWenTiLcof.java
package com.dbc;

public class QingWaTiaoTaiJieWenTiLcof {
    public int numWays(int n) {
        long pre = 1, cur = 1;
        for (int i = 2; i < n + 1; i++) {
            long temp = cur;
            cur = (cur + pre) % 1000000007;
            pre = temp;
        }
        return (int) cur;
    }
}
